inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java

```
@@ -0,0 +1,660 @@
+        tubeClientConfig.setSessionMaxAllowedDelayedMsgCount(sessionMaxAllowedDelayedMsgCount);
+        tubeClientConfig.setNettyWriteBufferHighWaterMark(nettyWriteBufferHighWaterMark);
+        tubeClientConfig.setHeartbeatPeriodMs(15000L);
+        tubeClientConfig.setRpcTimeoutMs(20000L);
+
+        return tubeClientConfig;
+    }
+
+    private void destroyConnection() {
+        for (Map.Entry<String, MessageProducer> entry : producerMap.entrySet()) {
+            MessageProducer producer = entry.getValue();
+            try {
+                producer.shutdown();
+            } catch (TubeClientException e) {
+                logger.error("destroy producer error in tubesink, MetaClientException {}", e.getMessage());
+            } catch (Throwable e) {
+                logger.error("destroy producer error in tubesink, ex {}", e.getMessage());
+            }
+        }
+        producerMap.clear();
+
+        if (sessionFactory != null) {
+            try {
+                sessionFactory.shutdown();
+            } catch (TubeClientException e) {
+                logger.error("destroy sessionFactory error in tubesink, MetaClientException {}",
+                        e.getMessage());
+            } catch (Exception e) {
+                logger.error("destroy sessionFactory error in tubesink, ex {}", e.getMessage());
+            }
+        }
+        sessionFactory = null;
+        logger.debug("closed meta producer");
+    }
+
+    /**
+     * Currently, all topics are published by the same producer. If needed, extend it to multi producers.
+     *
+     * @param topic
+     * @throws TubeClientException
+     */
+    private void initTopicProducer(String topic) throws TubeClientException {
+        if (StringUtils.isEmpty(topic)) {
+            logger.error("topic is empty");
+            return;
+        }
+        if (sessionFactory == null) {
+            logger.error("sessionFactory is null, can't create producer");
+            return;
```

Maybe need to throw an exception when the sessionFactory is null?